### PR TITLE
clean the gitRepo instead of the gitClient in updateconfig

### PR DESCRIPTION
**Pull request by hktime**
Fix [#18049](https://github.com/kubernetes/test-infra/issues/18049)

A small bug fixing, hopefully it will work :)

/hold

### CODE REVIEW

**Comment by hktime** [comment]
I signed it

File: prow/plugins/updateconfig/updateconfig_test.go [hunk 1]
```diff
@@ -1428,8 +1428,8 @@ func testUpdate(clients localgit.Clients, t *testing.T) {
 			t.Fatalf("Failed to clone: %v.", err)
 		}
 		defer func() {
-			if err := c.Clean(); err != nil {
```
Comment (alvaroaleman) [inline prow/plugins/updateconfig/updateconfig_test.go:1431]: In the test its actually correct to clean everything

Comment (hktime) [reply prow/plugins/updateconfig/updateconfig_test.go]: Exactly. Fixed.

**Pull request merged**